keanu-project/src/main/java/io/improbable/keanu/algorithms/variational/optimizer/gradient/GradientOptimizer.java

```diff
@@ -116,10 +123,10 @@ private double optimize(List<Vertex> outputVertices) {
         bayesianNetwork.cascadeObservations();
 
         FitnessFunctionWithGradient fitnessFunction = new FitnessFunctionWithGradient(
```

Comment:
It needs to be generic in order to specify that the `V` extends two different things: the `Vertex` class and the `Probabilistic` interface. I could revert it to taking a List of `Vertex`es instead - although this would mean that the variable name `probabilisticVertices` is misleading and should probably be changed to just `vertices`. Then the call to `Probabilistic.keepOnlyProbabilisticVertices` would happen later, inside the `FitnessFunctionWithGradient`.

Does that sound good?